src/WebJobs.Extensions.DurableTask/Scale/DurableTaskTriggersScaleProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#nullable enable
+#if FUNCTIONS_V3_OR_GREATER
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask.Scale
+{
+    internal class DurableTaskTriggersScaleProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private IScaleMonitor monitor;
+        private ITargetScaler targetScaler;
+
+        public DurableTaskTriggersScaleProvider(
+            IServiceProvider serviceProvider,
+            TriggerMetadata triggerMetadata)
+        {
+            string functionId = triggerMetadata.FunctionName;
+            FunctionName functionName = new FunctionName(functionId);
+
+            var options = this.GetOptions(serviceProvider, triggerMetadata);
+
+            var durabilityProviderFactory = this.GetDurabilityProviderFactory(serviceProvider, options);
+            var defaultDurabilityProvider = durabilityProviderFactory.GetDurabilityProvider();
+
+            var connectionName = durabilityProviderFactory is AzureStorageDurabilityProviderFactory azureStorageDurabilityProviderFactory
+                ? azureStorageDurabilityProviderFactory.DefaultConnectionName
+                : null;
+
+            var scaleUtils = new ScaleUtils();
+
+            this.targetScaler = scaleUtils.GetTargetScaler(
+                defaultDurabilityProvider,
+                functionId,
+                functionName,
+                connectionName,
+                options.HubName);
+
+            this.monitor = scaleUtils.GetScaleMonitor(
+                defaultDurabilityProvider,
+                functionId,
+                functionName,
+                connectionName,
+                options.HubName);
+        }
+
+        private DurableTaskOptions GetOptions(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata)
+        {
+            var nameResolver = serviceProvider.GetService<INameResolver>();
+
+            // the metadata is the sync triggers payload
+            var metadataString = triggerMetadata.Metadata.ToString();
+            var metadata = JsonConvert.DeserializeObject<DurableTaskMetadata>(metadataString);
```

Comment:
Is `triggerMetadata.Metadata` a `JObject`? If so, the more direct way to turn this into an instance of a class is to call `ToObject<T>()`
```suggestion
            var metadata = triggerMetadata.Metadata.ToObject<DurableTaskMetadata>();
````

Reply:
Cool trick. Incorporated on my next commit